### ISSUE TITLE
`compound-name` lint incorrectly warns for idiomatic attribute names with `as-*` and `is-*` prefixes

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#806]
The `compound-name` lint currently flags all compound attribute names, including idiomatic patterns that are widely used in EO's standard library:

- `as-*` for type conversions: `as-bytes`, `as-i64`, `as-i32`, `as-number`, `as-bool`
- `is-*` for boolean predicates: `is-empty`, `is-nan`, `is-finite`, `is-integer`, `is-windows`

These patterns are valuable conventions that:
1. Clearly communicate intent (conversion vs predicate)
2. Are consistent across the standard library
3. Follow common naming conventions in many languages

Currently, `eo-runtime` has 20 files with `+unlint compound-name` to suppress warnings for these idiomatic names.

**Proposal:**

Add an allowlist of idiomatic prefixes that should not trigger the warning:
- `as-` (type conversion)
- `is-` (boolean predicate)

The XSL function could be updated:
```xsl
<xsl:function name="eo:allowed-compound" as="xs:boolean">
  <xsl:param name="name"/>
  <xsl:sequence select="starts-with($name, 'as-') or starts-with($name, 'is-')"/>
</xsl:function>
```

And the condition updated to:
```xsl
<xsl:for-each select="//o[@base and @name and not(eo:special(@name)) and eo:compound(@name) and not(eo:allowed-compound(@name))]">
```

**Comment by yegor256** [comment]
@rultor release, tag is `0.1.0`

**Comment by yegor256** [comment]
@rultor release, tag is `0.1.0`

**Comment by yegor256** [comment]
@rultor release, tag is `0.1.0`

**Comment by yegor256** [comment]
@rultor release, tag is `0.1.0`

**Comment by yegor256** [comment]
@rultor release, tag is `0.1.0`

**Comment by 0crat** [comment]
@yegor256 Hey! Nice work closing that issue! 🎉 You've earned **+5 points** for this one: **+8** as the base reward, then **-3** for the 7 days it took to resolve (our policy deducts 0.5 points per day of delay). Your running score is now [+115](https://www.eolang.org/zerocracy/objectionary-vitals.html) - pretty solid! Don't forget to check your [Zerocracy account](https://www.zerocracy.com/) to see how you're doing overall.
